pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go

```diff
@@ -354,16 +357,24 @@ func (admitter *VMsAdmitter) authorizeVirtualMachineSpec(ar *admissionv1.Admissi
 			}
 		}
 
-		proxy := &authProxy{client: admitter.VirtClient, dataSourceInformer: admitter.DataSourceInformer, namespaceInformer: admitter.NamespaceInformer}
-		dv := &cdiv1.DataVolume{
-			ObjectMeta: dataVolume.ObjectMeta,
-			Spec:       dataVolume.Spec,
-		}
-		dv.Namespace = targetNamespace
-		allowed, message, err := admitter.cloneAuthFunc(dv, ar.Namespace, ar.Name, proxy, targetNamespace, serviceAccountName)
-		if err != nil && err != cdiv1.ErrNoTokenOkay {
+		allowed := true
+		message := ""
+		dv, err := storagetypes.GetDataVolumeFromCache(targetNamespace, dataVolume.Name, admitter.DataVolumeInformer.GetStore())
```

Comment:
Can we do a real request instead?

Reply:
TBH I am not sure what is best here - this would amplify the request
(many DVs for VM, though not common in practice so still constant).

This webhook runs on create/update but maybe this part is skipped if no spec change occurred,
if not, I think informer makes more sense

Reply:
Using informers for other types so being consistent with DataVolumes makes the most sense to me.  THe same auth check is done by virt controller so minor races here are fine this check is just for more obvious user feedback

Reply:
>This webhook runs on create/update but maybe this part is skipped if no spec change occurred,
if not, I think informer makes more sense

On update I would assume you could skip the check under some conditions...

>THe same auth check is done by virt controller so minor races here are fine this check is just for more obvious user feedback

Sounds good.

Reply:
@xpivarc adding a commit to skip auth if dataVolumeTemplates unchanged on update